### PR TITLE
Translate Linux fsflags to BSD flags and vice versa

### PR DESCRIPTION
**Pull request by enkore**
Another leftover from weeks ago.

This also adds some output in the py.test header about the test platform, we have quite a few possibilities there and it is not immediately obvious.

Implements the popular flags from  #618. Many flags are read-only (informational) or generally not very useful or unsupported. The selection made here is relatively portable and useful. Linux-specific flags should not be ORed into bsdflags.

### CODE REVIEW

File: borg/archive.py [hunk 1]
```diff
@@ -25,14 +25,13 @@
     CompressionDecider1, CompressionDecider2, CompressionSpec, \
     IntegrityError
 from .repository import Repository
-from .platform import acl_get, acl_set
+from .platform import acl_get, acl_set, change_flags, get_flags
```
Comment (ThomasWaldmann) [inline borg/archive.py]: set_flags?

File: borg/platform_linux.pyx [hunk 2]
```diff
@@ -33,10 +34,72 @@ cdef extern from "fcntl.h":
...
+    stat.UF_COMPRESSED: FS_COMPR_FL,
+}
+
+
+def change_flags(path, bsd_flags, fd=None):
+    cdef int flags = 0
+    for bsd_flag, linux_flag in BSD_TO_LINUX_FLAGS.items():
+        if bsd_flags & bsd_flag:
+            flags |= linux_flag
+    if stat.S_ISLNK(os.lstat(path).st_mode):
```
Comment (ThomasWaldmann) [inline borg/platform_linux.pyx]: move this above the for-loop?

File: borg/platform_linux.pyx [hunk 3]
```diff
@@ -33,10 +34,72 @@ cdef extern from "fcntl.h":
...
+        return
+    open_fd = not fd
```
Comment (ThomasWaldmann) [inline borg/platform_linux.pyx]: fd is None

File: borg/testsuite/archiver.py [hunk 4]
```diff
@@ -26,15 +26,13 @@
...
 from ..repository import Repository
-from . import BaseTestCase, changedir, environment_variable
+from .conftest import has_lchflags, has_llfuse
+from . import BaseTestCase, changedir, environment_variable, conftest
 
 try:
     import llfuse
-    has_llfuse = True or llfuse  # avoids "unused import"
 except ImportError:
-    has_llfuse = False
```
Comment (ThomasWaldmann) [inline borg/testsuite/archiver.py]: why did you change this?

File: borg/testsuite/conftest.py [hunk 5]
```diff
@@ -2,3 +2,38 @@
...
+except OSError:
+    has_lchflags = False
+    no_lchlfags_because = '(the file system at %s does not support flags)' % tempfile.gettempdir()
+
+
+try:
+    import llfuse
+    has_llfuse = True or llfuse  # avoids "unused import"
+except ImportError:
+    has_llfuse = False
```
Comment (ThomasWaldmann) [inline borg/testsuite/conftest.py]: ah, here it is.

File: borg/testsuite/conftest.py [hunk 6]
```diff
@@ -2,3 +2,38 @@
...
+
+
+def pytest_report_header(config, startdir):
+    yesno = ['no', 'yes']
+    flags = 'Testing flags: %s %s' % (yesno[has_lchflags], no_lchlfags_because)
```
Comment (ThomasWaldmann) [inline borg/testsuite/conftest.py]: hmm, just "flags" is hard to understand out of context.

**Comment by ThomasWaldmann** [comment]
did you test on *bsd?

**Comment by enkore** [comment]
No, but the same logic applies to !Linux. Testing it on any of OSX or on a BSD is sufficient to check whether the correct logic still applies.

**Comment by enkore** [comment]
Testing on Vagrant netbsd64

```
==> netbsd64: Testing BSD-style flags: yes 
==> netbsd64: Testing fuse: no
==> netbsd64: fakeroot: no (>=1.20.2: no)
==> netbsd64: ============ 506 passed, 140 skipped, 2 xfailed in 1046.27 seconds =============
```

There where some changes with the conftest stuff, I moved it to the root directory so that it works with --pyargs correctly.
